src/code/z_bgcheck.c

```diff
@@ -4287,21 +4287,18 @@ s32 WaterBox_GetSurface2(PlayState* play, CollisionContext* colCtx, Vec3f* pos,
         waterBox = &colHeader->waterBoxes[i];
 
         room = WATERBOX_ROOM(waterBox->properties);
-        if (!(room == play->roomCtx.curRoom.num || room == WATERBOX_ROOM_ALL)) {
-            continue;
-        }
-        if (waterBox->properties & WATERBOX_FLAG_19) {
-            continue;
-        }
-        if (!(waterBox->xMin < pos->x && pos->x < waterBox->xMin + waterBox->xLength)) {
-            continue;
-        }
-        if (!(waterBox->zMin < pos->z && pos->z < waterBox->zMin + waterBox->zLength)) {
-            continue;
-        }
-        if (pos->y - surfaceChkDist < waterBox->ySurface && waterBox->ySurface < pos->y + surfaceChkDist) {
-            *outWaterBox = waterBox;
-            return i;
+        if (room == play->roomCtx.curRoom.num || room == WATERBOX_ROOM_ALL) {
+            if (!(waterBox->properties & WATERBOX_FLAG_19)) {
+                if (waterBox->xMin < pos->x && pos->x < waterBox->xMin + waterBox->xLength) {
+                    if (waterBox->zMin < pos->z && pos->z < waterBox->zMin + waterBox->zLength) {
+                        if (pos->y - surfaceChkDist < waterBox->ySurface &&
+                            waterBox->ySurface < pos->y + surfaceChkDist) {
+                            *outWaterBox = waterBox;
+                            return i;
+                        }
+                    }
+                }
+            }
```

Comment:
ngl the continues are kinda easier to read. im not totally opposed to nesting it with ifs, but yeah would be interested in other thoughts

Reply:
this is to make it more similar to `WaterBox_GetSurfaceImpl` and `func_800425B0` (I don't think it's a bold claim to say copypaste has been used here)

I'd be ok with continues too, just that I would like to keep the three funcs consistent

Reply:
or

```c
        if ((room == play->roomCtx.curRoom.num || room == WATERBOX_ROOM_ALL) &&
            !(waterBox->properties & WATERBOX_FLAG_19) && waterBox->xMin < pos->x &&
            pos->x < waterBox->xMin + waterBox->xLength && waterBox->zMin < pos->z &&
            pos->z < waterBox->zMin + waterBox->zLength && pos->y - surfaceChkDist < waterBox->ySurface &&
            waterBox->ySurface < pos->y + surfaceChkDist) {

            *outWaterBox = waterBox;
            return i;
        }
```

```c
        if (room == play->roomCtx.curRoom.num || room == WATERBOX_ROOM_ALL) {
            if (!(waterBox->properties & WATERBOX_FLAG_19)) {
                if (waterBox->xMin < pos->x && pos->x < waterBox->xMin + waterBox->xLength && waterBox->zMin < pos->z &&
                    pos->z < waterBox->zMin + waterBox->zLength && pos->y - surfaceChkDist < waterBox->ySurface &&
                    waterBox->ySurface < pos->y + surfaceChkDist) {

                    *outWaterBox = waterBox;
                    return i;
                }
            }
        }
```

...

Reply:
I actually think I prefer the continues here as well, I'm sort of ashamed to admit, with the single if statement getting my second choice vote.

though fwiw I also prefer a different boolean composition style with the operators at the start of the line, so maybe my preference only makes sense with that in mind
```c
if ((room == play->roomCtx.curRoom.num || room == WATERBOX_ROOM_ALL)
    && !(waterBox->properties & WATERBOX_FLAG_19)
    && waterBox->xMin < pos->x
    && pos->x < waterBox->xMin + waterBox->xLength
    && waterBox->zMin < pos->z
    && pos->z < waterBox->zMin + waterBox->zLength
    && pos->y - surfaceChkDist < waterBox->ySurface
    && waterBox->ySurface < pos->y + surfaceChkDist
) {
```

Reply:
Yeah that would be my preferred formatting style as well (operators aligned to the right) but it's not what our formatter config does, if it's even capable of doing that :(